src/main/scala/ai/privado/languageEngine/java/tagger/sink/JavaAPITagger.scala

```diff
@@ -0,0 +1,148 @@
+    .l
+
+  lazy val apiTaggerToUse = apis.methodFullName(s"${commonHttpPackages}${APISINKS_REGEX}.*").l.size match {
+    case 0 =>
+      if (httpPackagesInImport()) {
+        MetricHandler.metricsData("apiTaggerVersion") = Json.fromString(APITaggerVersionJava.V1Tagger.toString)
+        APITaggerVersionJava.V1Tagger
+      } else {
+        MetricHandler.metricsData("apiTaggerVersion") = Json.fromString(APITaggerVersionJava.SkipTagger.toString)
+        APITaggerVersionJava.SkipTagger
+      }
+    case _ =>
+      MetricHandler.metricsData("apiTaggerVersion") = Json.fromString(APITaggerVersionJava.V2Tagger.toString)
+      APITaggerVersionJava.V2Tagger
+  }
+
+  implicit val engineContext: EngineContext = EngineContext(semantics = getDefaultSemantics, config = EngineConfig(4))
+  val commonHttpPackages: String =
+    "^(?i)(org.apache.http|okhttp|org.glassfish.jersey|com.mashape.unirest|java.net.http|java.net.URL|org.springframework.(web|core.io)|groovyx.net.http|org.asynchttpclient|kong.unirest.java|org.concordion.cubano.driver.http|javax.net.ssl).*"
+  override def run(builder: BatchedUpdate.DiffGraphBuilder): Unit = {
+    val apiInternalSinkPattern = cpg.literal.code("(?:\"|')(" + ruleInfo.combinedRulePattern + ")(?:\"|')").l
+    val propertySinks          = cpg.property.filter(p => p.value matches (ruleInfo.combinedRulePattern)).usedAt.l
+    apiTaggerToUse match {
+      case APITaggerVersionJava.V1Tagger =>
+        logger.debug("Using brute API Tagger to find API sinks")
+        sinkTagger(apiInternalSinkPattern, apis, builder, ruleInfo)
+        sinkTagger(propertySinks, apis, builder, ruleInfo)
+      case APITaggerVersionJava.V2Tagger =>
+        logger.debug("Using Enhanced API tagger to find API sinks")
+        sinkTagger(apiInternalSinkPattern, apis.methodFullName(commonHttpPackages).l, builder, ruleInfo)
+        sinkTagger(propertySinks, apis.methodFullName(commonHttpPackages).l, builder, ruleInfo)
+      case _ =>
+        logger.debug("Skipping API Tagger because valid match not found")
+    }
+  }
+
+  private def sinkTagger(
+    apiInternalSinkPattern: List[CfgNode],
+    apis: List[CfgNode],
+    builder: BatchedUpdate.DiffGraphBuilder,
+    ruleInfo: RuleInfo
+  ): Unit = {
+    val filteredLiteralSourceNode = apiInternalSinkPattern.filter(node => isFileProcessable(getFileNameForNode(node)))
+    if (apis.nonEmpty && filteredLiteralSourceNode.nonEmpty) {
+      val apiFlows = apis.reachableByFlows(filteredLiteralSourceNode).l
+      apiFlows.foreach(flow => {
+        val literalCode = flow.elements.head.originalPropertyValue.getOrElse(flow.elements.head.code)
+        val apiNode     = flow.elements.last
+        addRuleTags(builder, apiNode, ruleInfo)
+        storeForTag(builder, apiNode)(Constants.apiUrl, literalCode)
+      })
+    }
+  }
+
+  private def httpPackagesInImport(): Boolean = {
+    val imports = ImportUtility.getAllImportsFromProject(AppCache.localScanPath, Language.JAVA)
```

Comment:
@abhstabs We need to check if http packages are present in the import statements, currently we are not doing that.

ImportUtility.getAllImportsFromProject ---> this returns all the imports